Allow websocket URL to contain query string params

jatt opened the pull request:
Refactored WebSocketService#normalizeURL method to handle URL with query params correctly (add / before ?, not to the end of the URL).

thoov commented on app/services/websockets.js:
```diff
@@ -85,11 +85,19 @@ export default Ember.Service.extend({
 
-    if(parsedUrl.path() === '/' && url.slice(-1) !== '/') {
-      return url + '/';
+    if(parsedUrl.path() !== '/') {
```
Just to clean this up lets make the logic like this:

```js
normalizeURL(url) {
  var parsedUrl = new URI(url);
  var path = parsedUrl.path();
  var query = parsedUrl.query();

  if(path !== '/' && query === '' && url.slice(-1) !== '/') {
    return url + '/'';
  }
  else if(url.indexOf('/?') === -1) {
    return url.replace('?', '/?');
  }

  return url;
}

thoov commented:
Thanks @jatt for the PR! Once you address the comment above I will merge

thoov commented:
We also need to update this: https://github.com/thoov/ember-websockets/blob/master/app/services/websockets.js#L85-L93

I would suggest refactoring the code to use a common function in both places.

jatt commented:
I refactored this normalizeURL function into a separate mixin that is included into all 3 services. Also I cleaned up a method a bit, but it is now as simple as it can be, your code doesn't cover all cases and tests fail.
